ethminer/MinerAux.h

```diff
@@ -308,7 +308,14 @@ class MinerCLI
 #if ETH_ETHASHCL || ETH_ETHASHCUDA
 		else if ((arg == "--cl-global-work" || arg == "--cuda-grid-size")  && i + 1 < argc)
```

Comment:
Just split this up into 2 if's. Someone just tried a shortcut here and overloaded a variable with two distinct meanings. You did the right thing creating two separate unrelated control vars. Now handle them in separate ifs.